repose-aggregator/extensions/simple-rbac/src/main/scala/org/openrepose/filters/simplerbac/SimpleRbacFilter.scala

```diff
@@ -0,0 +1,268 @@
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.apache.commons.lang3.StringUtils
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.commons.utils.StringUriUtilities
+import org.openrepose.commons.utils.servlet.http.{MutableHttpServletRequest, MutableHttpServletResponse}
+import org.openrepose.core.filter.FilterConfigHelper
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.core.spring.ReposeSpringProperties
+import org.openrepose.filters.apivalidator.DispatchHandler
+import org.openrepose.filters.simplerbac.config.SimpleRbacConfig
+import org.springframework.beans.factory.annotation.Value
+
+import scala.collection.JavaConversions._
+import scala.io.Source
+import scala.util.Try
+
+@Named
+class SimpleRbacFilter @Inject()(configurationService: ConfigurationService,
+                                 @Value(ReposeSpringProperties.CORE.CONFIG_ROOT) configurationRoot: String)
+
+  extends Filter
+  with UpdateListener[SimpleRbacConfig]
+  with LazyLogging {
+
+  private final val DEFAULT_CONFIG = "simple-rbac.cfg.xml"
+
+  var configurationFile: String = DEFAULT_CONFIG
+  var configuration: SimpleRbacConfig = _
+  var initialized = false
+  var validator: Validator = _
+  val config = new Config
+
+  override def init(filterConfig: FilterConfig): Unit = {
+    configurationFile = new FilterConfigHelper(filterConfig).getFilterConfig(DEFAULT_CONFIG)
+    logger.info("Initializing filter using config " + configurationFile)
+    val xsdURL: URL = getClass.getResource("/META-INF/schema/config/simple-rbac.xsd")
+    configurationService.subscribeTo(
+      filterConfig.getFilterName,
+      configurationFile,
+      xsdURL,
+      this,
+      classOf[SimpleRbacConfig]
+    )
+  }
+
+  override def destroy(): Unit = {
+    configurationService.unsubscribeFrom(configurationFile, this)
+  }
+
+  override def doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain): Unit = {
+    if (!initialized) {
+      logger.error("Simple RBAC filter has not yet initialized...")
+      servletResponse.asInstanceOf[HttpServletResponse].sendError(500)
+    } else {
+      val mutableHttpRequest = MutableHttpServletRequest.wrap(servletRequest.asInstanceOf[HttpServletRequest])
+      val mutableHttpResponse = MutableHttpServletResponse.wrap(mutableHttpRequest, servletResponse.asInstanceOf[HttpServletResponse])
+
+      logger.trace("Simple RBAC filter processing request...")
+      validator.validate(mutableHttpRequest,mutableHttpResponse,filterChain)
+    }
+    logger.trace("Simple RBAC filter returning response...")
+  }
+
+  override def configurationUpdated(configurationObject: SimpleRbacConfig): Unit = {
+    configuration = configurationObject
+    val isDelegating = configuration.getDelegating != null
+    val delegationQuality = if (isDelegating) configuration.getDelegating.getQuality else 0.0
```

Comment:
I didn't like that when I wrote it, but I couldn't make it look good. With your suggestions and a fresh set of eyes, I think I may have cleaned this spot up fairly well.